modules/launcher/src/test/java/org/mule/module/launcher/DeploymentServiceTestCase.java

```diff
@@ -2865,6 +2867,37 @@ private void assertApplicationDeploymentSuccess(DeploymentListener listener, Str
         assertDeploymentSuccess(listener, artifactName);
         assertStatus(artifactName, ApplicationStatus.STARTED);
     }
+    
+    private void assertApplicationDeploymentSuccessMulltipleArtifacts(DeploymentListener listener, String... artifactNames)
+    {
+        assertDeploytmentSuccessMultipleArtifacts(listener, artifactNames);
+        for (String artifactName : artifactNames)
+        {
+            assertStatus(artifactName, ApplicationStatus.STARTED);
+        }
+    }
+
+    private void assertDeploytmentSuccessMultipleArtifacts(final DeploymentListener listener, final String... artifactNames)
+    {
+        Prober prober = new PollingProber(DEPLOYMENT_TIMEOUT, 100);
+        prober.check(new JUnitProbe()
+        {
+            @Override
+            protected boolean test() throws Exception
+            {
+                ArgumentCaptor<String> arguments = forClass(String.class);
+                verify(listener, times(artifactNames.length)).onDeploymentSuccess(arguments.capture());
```

Comment:
by default mockito doesn't verify the order of the invocation (there is another construct to do that specifically),  so i don't think that's the cause of this flakyness.